testsuite/elytron/src/test/java/org/wildfly/test/security/common/AbstractConfigSetupTask.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.test.security.common;
+
+import java.util.Arrays;
+import java.util.ListIterator;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.jboss.as.test.integration.management.util.ServerReload;
+import org.jboss.logging.Logger;
+import org.wildfly.test.security.common.elytron.ConfigurableElement;
+
+/**
+ * Abstract parent for server setup tasks configuring. Implementing classes overrides {@link #getConfigurableElements()}
+ * method to provide configured on which this task will call {@link ConfigurableElement#create(CLIWrapper)}.
+ *
+ * @author Josef Cacek
+ */
+public abstract class AbstractConfigSetupTask {
+
+    private static final Logger LOGGER = Logger.getLogger(AbstractConfigSetupTask.class);
+
+    private ConfigurableElement[] configurableElements;
+
+    /**
+     * Creates configuration elements (provided by implementation of {@link #getConfigurableElements()} method) and calls
+     * {@link ConfigurableElement#create(CLIWrapper)} for them.
+     */
+    protected void setup(final ModelControllerClient modelControllerClient) throws Exception {
+        configurableElements = getConfigurableElements();
+
+        if (configurableElements == null || configurableElements.length == 0) {
+            LOGGER.warn("Empty Elytron configuration.");
+            return;
+        }
+
+        try (CLIWrapper cli = new CLIWrapper(true)) {
+            for (final ConfigurableElement configurableElement : configurableElements) {
+                LOGGER.infov("Adding element {0} ({1})", configurableElement.getName(),
```

Comment:
Maybe this should be `debug` so we don't use up too much space on CI with logs. Not a huge deal though.